### PR TITLE
Fixes for BUILD_ATEN_ONLY build flavor

### PR DESCRIPTION
**Pull request by sinkingsugar**
Very small fixes to make BUILD_ATEN_ONLY finally link properly when used as dynamic library.

Our daily CI failed to link, looks like some stuff in `caffe2/core/allocator.cc` was missing, not sure why it didn't pick it up before. I guess that yesterday big changes triggered it finally!

### CODE REVIEW

File: aten/src/ATen/native/Indexing.cpp [hunk 1]
```diff
@@ -282,8 +282,10 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
                   sourceSlicedSizes.begin())) {
     std::stringstream ss;
     ss << "index_copy_(): Source/destination tensor must have same slice shapes. ";
+#ifndef BUILD_ATEN_ONLY
```
Comment (ezyang) [inline aten/src/ATen/native/Indexing.cpp]: Errr... why an ifdef here?

**Comment by ezyang** [comment]
This isn't really sustainable, but I guess it will get solved properly once c10/ folder absorbs more stuff. I am a bit perplexed about the ifdef in the printing code though. What's going on there?

File: aten/src/ATen/native/Indexing.cpp (hunk 1 above)
Comment (sinkingsugar) [reply aten/src/ATen/native/Indexing.cpp]: Well because:

**Comment by sinkingsugar** [comment]
I think C++ is going on :) 
I didn't really focus on finding it cos... no time yet but I will if you are concerned, altho I suspect you guys will refactor and track those headers at some point with c10 changes.

**Pull request closed without merging**